pkgs/development/tools/build-managers/xmake/default.nix

```diff
@@ -1,48 +1,30 @@
-{ lib
-, stdenv
-, fetchurl
-, pkg-config
-, lua
-, readline
-, ncurses
-, lz4
-, tbox
-, xmake-core-sv
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  CoreServices,
+  ...
 }:
-
 stdenv.mkDerivation rec {
   pname = "xmake";
-  version = "2.9.4";
-
-  src = fetchurl {
-    url = "https://github.com/xmake-io/xmake/releases/download/v${version}/xmake-v${version}.tar.gz";
-    hash = "sha256-deLd4r0qSKMymJuAGuZQd8RS1JH+xRep2yeoHIcTzcU=";
+  version = "v2.9.5";
+  src = fetchFromGitHub {
+    owner = "xmake-io";
+    repo = "xmake";
+    rev = version;
+    hash = "sha256-/hzYrdB+0up4o78zE1KkhQY0N1trnd9fcrp7s/spVfo=";
+    fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
-
-  buildInputs = [
-    lua
```

Comment:
Whenever possible, we should use build dependencies provided by nixpkgs instead of project‘s submodules.

Reply:
Emmmm, that's an important point.

Reply:
tbox and xmake-core-sv are only used by xmake, so they should be upgraded to the versions required by xmake. If there are important bug fixes, you can also temporarily upgrade them to the unstable version.

Reply:
> Whenever possible, we should use build dependencies provided by nixpkgs instead of project‘s submodules.

Then at least we need to build 7 extra packages for (and, only for) xmake, according to its [.gitsubmodule](https://github.com/xmake-io/xmake/blob/dev/.gitmodules)? Many dependencies of xmake are just forks of upstream projects with some specialized patches.

Reply:
The upstream author said that forking is only for submodules to facilitate git submodules management and fix it to a specific commit. Usually, no code of third-party dependencies will be changed.

https://github.com/xmake-io/xmake/discussions/3107